src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,130 @@
+  flexWrap: 'wrap',
+  paddingBottom: '10px',
+  gap: 8
+}));
+
+const ShowMore = styled(Button, {
+  name: 'ESChips',
+  slot: 'showMore',
+  overridesResolver: (_, styles) => styles.showMore
+})(() => ({
+  flexShrink: 0
+}));
+
+export const MChip = styled(Chip)(({ theme }) => ({
+  backgroundColor: theme.palette.monoA.A50,
+  fontSize: '12px'
+}));
+
+export const Chips = (inProps: ChipProps) => {
+  const [hiddenItems, setHiddenItems] = useState<Record<number, boolean>>({});
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [showMore, setShowMore] = useState(false);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const { children, labelButtonMore } = useThemeProps({ props: inProps, name: 'ESChips' });
+
+  useResizeObserver(containerRef, (entries: any) => {
+    const entry = entries[0];
+    if (containerWidth === entry.contentRect.width) {
+      return;
+    }
+    setHiddenItems({});
+    setContainerWidth(entry.contentRect.width);
+  });
+
+  useLayoutEffect(() => {
+    const containerElement = containerRef.current;
+    const containerChildren = containerElement?.children;
+    if (!showMore) {
+      if (!containerChildren || !containerWidth) {
+        return;
+      }
+      containerElement.style.flexWrap = 'nowrap';
+
+      const elementWidths: number[] = [];
+      let showDonw = 0;
+
+      for (let i = 0, l = containerChildren.length; i < l; i++) {
+        const child = containerChildren[i];
+        (child as HTMLElement).style.display = 'flex';
+        const clientRect = child.getBoundingClientRect();
+
+        const { width } = clientRect;
+
+        const totoalWidth = i > 0 ? 8 + width : width;
+
+        if (child.getAttribute('data-element') === 'more') {
+          showDonw = totoalWidth;
+        } else {
+          elementWidths.push(totoalWidth);
+        }
+      }
+
+      const hiddenItemsMap: Record<number, boolean> = {};
+      let remaningContainerWidth = containerWidth - showDonw;
+
+      Array.from(containerChildren).forEach((item, index: number) => {
+        const itemWidth = elementWidths[index];
+        if (itemWidth <= remaningContainerWidth) {
+          remaningContainerWidth -= itemWidth;
+        } else if (item.getAttribute('data-element') === 'more') {
+          (item as HTMLElement).style.display = 'flex';
+        } else {
+          (item as HTMLElement).style.display = 'none';
+          hiddenItemsMap[+(item as HTMLElement).id] = true;
+        }
+      });
+      setHiddenItems(hiddenItemsMap);
+    } else {
+      if (!containerChildren || !containerWidth) {
+        return;
+      }
+      containerElement.style.flexWrap = 'wrap';
+      Array.from(containerChildren).forEach((item) => {
+        (item as HTMLElement).style.display = 'flex';
+      });
+    }
+  }, [children, containerWidth, showMore]);
+
+  const toggleShowMore = () => {
+    setShowMore(!showMore);
+  };
+
+  return (
+    <StackRoot ref={containerRef} direction="row" spacing={1}>
```

Comment:
Можно уже просто сделать `styled('div')` и
```suggestion
    <StackRoot ref={containerRef}>
```